pkg/core/identity.go

```diff
@@ -169,12 +171,25 @@ func (i *IdentityBase) Validate(ctx context.Context) (err error) {
 	if err = ValidateFFNameFieldNoUUID(ctx, i.Namespace, "namespace"); err != nil {
 		return err
 	}
-	if err = ValidateFFNameFieldNoUUID(ctx, i.Name, "name"); err != nil {
+
+	options := NameValidationOptions{noUUID: true}
+	if i.Type == IdentityTypeCustom {
+		options.extraAllowedChars = "/"
+	}
+	if err = ValidateFFNameFieldOptions(ctx, i.Name, "name", options); err != nil {
 		return err
 	}
+
+	var legacyDID string
+	if i.Type == IdentityTypeCustom {
+		if strings.HasPrefix(i.DID, FireFlyNodeDIDPrefix) || strings.HasPrefix(i.DID, FireFlyOrgDIDPrefix) {
```

Comment:
As noted on the FIR, this does mean we can never add additional "reserved" prefixes. I'm on the fence about whether that's a good move, so open to thoughts on whether we should have some other well-defined prefix for custom identities.

Reply:
What about _not_ allowing `/` in custom ones? Wouldn't that solve it.

- `did:firefly:bob/sally` - this is only acceptable if `bob` is an understood category
- `did:firefly:sally` - this is a custom one, because it does not allow `/`